## Q1

In [2]:
! mlflow --version

mlflow, version 2.3.1


## Q2

In [1]:
! python preprocess_data.py --raw_data_path="../../data/" --dest_path ./output

In [3]:
import os

dv_size_bytes = os.path.getsize("output/dv.pkl")

dv_size_kb = (dv_size_bytes * 8) * 1e-3

print("Size in kilobit:", dv_size_kb, "kb")
print("Size in Megabit:", dv_size_kb * 1e-3, "Mb")
print("Size in kilobyte:", dv_size_kb / 8, "kB")

Size in kilobit: 1229.28 kb
Size in Megabit: 1.22928 Mb
Size in kilobyte: 153.66 kB


## Q3

```bash
$ python train.py --data_path="./output"
```

| Name|	Value |
| --- | --- |
| bootstrap|	True |
| ccp_alpha|	0.0 |
| criterion|	squared_error |
| max_depth|	10 |
| max_features|	1.0 |
| max_leaf_nodes|	None |
| max_samples|	None |
| min_impurity_decrease|	0.0 |
| min_samples_leaf|	1 |
| min_samples_split|	2 |
| min_weight_fraction_leaf|	0.0 |
| n_estimators|	100 |
| n_jobs|	None |
| oob_score|	False |
| random_state|	0 |
| train-data-path|	./output/train.pkl |
| valid-data-path|	./output/val.pkl |
| verbose|	0 |
| warm_start|	False |

## Launch the tracking server locally for MLflow

```bash
$ mlflow ui --backend-store-uri sqlite:///mlflow.db --artifacts-destination artifacts/
```


## Q4

In [61]:
#! python hpo.py

2023/05/22 13:16:38 INFO mlflow.tracking.fluent: Experiment with name 'random-forest-hyperopt' does not exist. Creating a new experiment.
[I 2023-05-22 13:16:38,967] A new study created in memory with name: no-name-6eaf96a7-becb-4011-be8c-4fd2d16c108a
[I 2023-05-22 13:16:39,939] Trial 0 finished with value: 2.451379690825458 and parameters: {'n_estimators': 25, 'max_depth': 20, 'min_samples_split': 8, 'min_samples_leaf': 3}. Best is trial 0 with value: 2.451379690825458.
[I 2023-05-22 13:16:40,612] Trial 1 finished with value: 2.4667366020368333 and parameters: {'n_estimators': 16, 'max_depth': 4, 'min_samples_split': 2, 'min_samples_leaf': 4}. Best is trial 0 with value: 2.451379690825458.
[I 2023-05-22 13:16:41,450] Trial 2 finished with value: 2.449827329704216 and parameters: {'n_estimators': 34, 'max_depth': 15, 'min_samples_split': 2, 'min_samples_leaf': 4}. Best is trial 2 with value: 2.449827329704216.
[I 2023-05-22 13:16:41,930] Trial 3 finished with value: 2.460983516558473 a

Best rmse: 2.449827

## Q5

```bash
$ python register_model.py
Registered model 'rf_best_test' already exists. Creating a new version of this model...
2023/05/22 13:26:37 INFO mlflow.tracking._model_registry.client: Waiting up to 300 seconds for model version to finish creation. Model name: rf_best_test, version 2
Created version '2' of model 'rf_best_test'.
```

best test rmse: 2.285

## Q6

- Version number
- Source run (is it the same as Source experiment?)
- Registered at
- Last modified
- ~~Model signature~~ (did not find it)